File: Back-end/src/main/java/KJX/KJX/controller/EventController.java
package KJX.KJX.controller;

import KJX.KJX.entity.Event;
import KJX.KJX.service.EventService;
import org.springframework.beans.factory.annotation.Autowired;
import org.springframework.http.ResponseEntity;
import org.springframework.web.bind.annotation.*;
import java.util.List;

@RestController
@RequestMapping("/api/events")
@CrossOrigin(origins = "*")
public class EventController {
    
    @Autowired
    private EventService eventService;
    
    @GetMapping
    public ResponseEntity<List<Event>> getAllEvents() {
        return ResponseEntity.ok(eventService.getAllEvents());
    }
    
    @GetMapping("/{id}")
    public ResponseEntity<?> getEventById(@PathVariable Long id) {
        return eventService.getEventById(id)
            .map(ResponseEntity::ok)
            .orElseGet(() -> ResponseEntity.notFound().build());
    }
    
    @GetMapping("/type/{type}")
    public ResponseEntity<List<Event>> getEventsByType(@PathVariable String type) {
        return ResponseEntity.ok(eventService.getEventsByType(type));
    }
    
    @GetMapping("/upcoming")
    public ResponseEntity<List<Event>> getUpcomingEvents() {
        return ResponseEntity.ok(eventService.getUpcomingEvents());
    }
    
    @GetMapping("/team/{teamId}")
    public ResponseEntity<List<Event>> getEventsByTeamId(@PathVariable Long teamId) {
        return ResponseEntity.ok(eventService.getEventsByTeamId(teamId));
    }
    
    @PostMapping
    public ResponseEntity<Event> createEvent(@RequestBody Event event) {
        return ResponseEntity.ok(eventService.saveEvent(event));
    }
    
    @PutMapping("/{id}")
    public ResponseEntity<Event> updateEvent(@PathVariable Long id, @RequestBody Event eventDetails) {
        return ResponseEntity.ok(eventService.updateEvent(id, eventDetails));
    }
    
    @DeleteMapping("/{id}")
    public ResponseEntity<Void> deleteEvent(@PathVariable Long id) {
        eventService.deleteEvent(id);
        return ResponseEntity.noContent().build();
    }
}

File: nodejs-backend/src/controllers/event.controller.js
/**
 * Event Controller
 * Handles event management endpoints
 * Converted from Spring Boot EventController
 */

const eventService = require('../services/event.service');

class EventController {
  /**
   * GET /api/events
   * Get all events
   */
  async getAllEvents(req, res, next) {
    try {
      const events = await eventService.getAllEvents();
      res.json(events);
    } catch (error) {
      next(error);
    }
  }

  /**
   * GET /api/events/:id
   * Get event by ID
   */
  async getEventById(req, res, next) {
    try {
      const event = await eventService.getEventById(req.params.id);
      if (!event) {
        return res.status(404).json({
          success: false,
          message: 'Event not found'
        });
      }
      res.json(event);
    } catch (error) {
      next(error);
    }
  }

  /**
   * GET /api/events/type/:type
   * Get events by type
   */
  async getEventsByType(req, res, next) {
    try {
      const events = await eventService.getEventsByType(req.params.type);
      res.json(events);
    } catch (error) {
      next(error);
    }
  }

  /**
   * GET /api/events/upcoming
   * Get upcoming events
   */
  async getUpcomingEvents(req, res, next) {
    try {
      const events = await eventService.getUpcomingEvents();
      res.json(events);
    } catch (error) {
      next(error);
    }
  }

  /**
   * GET /api/events/team/:teamId
   * Get events by team ID
   */
  async getEventsByTeamId(req, res, next) {
    try {
      const events = await eventService.getEventsByTeamId(req.params.teamId);
      res.json(events);
    } catch (error) {
      next(error);
    }
  }

  /**
   * POST /api/events
   * Create new event
   */
  async createEvent(req, res, next) {
    try {
      const event = await eventService.createEvent(req.body);
      res.json(event);
    } catch (error) {
      next(error);
    }
  }

  /**
   * PUT /api/events/:id
   * Update event
   */
  async updateEvent(req, res, next) {
    try {
      const event = await eventService.updateEvent(req.params.id, req.body);
      res.json(event);
    } catch (error) {
      if (error.message === 'Event not found') {
        return res.status(404).json({
          success: false,
          message: error.message
        });
      }
      next(error);
    }
  }

  /**
   * DELETE /api/events/:id
   * Delete event
   */
  async deleteEvent(req, res, next) {
    try {
      await eventService.deleteEvent(req.params.id);
      res.status(204).send();
    } catch (error) {
      if (error.message === 'Event not found') {
        return res.status(404).json({
          success: false,
          message: error.message
        });
      }
      next(error);
    }
  }
}

module.exports = new EventController();
